src/controller/CommissioningDelegate.h

```diff
@@ -376,12 +376,9 @@ class CommissioningParameters
         return *this;
     }
 
-    // If a ThreadOperationalDataset is provided, then the ThreadNetworkScan will not be attempted
     CommissioningParameters & SetThreadOperationalDataset(ByteSpan threadOperationalDataset)
     {
-
         mThreadOperationalDataset.SetValue(threadOperationalDataset);
```

Comment:
once you set mThreadOperationalDataset, it looks like mAttemptThreadNetworkScan cannot be changed anymore: SetAttemptThreadNetworkScan has a `if (!mThreadOperationalDataset.HasValue())` (which I find odd ... a setter that refuses to set.

I am not familiar with this code, but could you explain why mAttemptThreadNetworkScan exists? What is its purpuse and how does it achieve it? Maybe @chrisdecenzo too.

Overall this seems to be very suspect code:
  - a `Optional<bool>` that is only ever used as a bool. It should be a bool initialized to false it seems
  - a setter that conditionally does not set, which I would count as a bug unless there is a serious reason for it ... and then the method should not be called `set` or some other workaround should be done

Please try to explain me the logic, since I am surprised I have so much trouble understanding the flow given that this is a simple member variable with maybe 5 usages :(

Reply:
Only java seems to use the `SetAttemptThreadNetworkScan`. Why not drop it entirely then if we always want to scan?

Reply:
Same questions for attempted wifi scans - why optional<bool> and why only java? At least the setter on that one does not have conditionals ...

Reply:
I suggest the following changes:

- [ ] remove conditional from the setters
- [ ] make these variables bool (not optional)
- [ ] remove this line as well as the PR does

Also consider why java needs to alter these but other code does not currently. This should be documented and I saw no clear documentation. Maybe these variables should not exist.

Reply:
@andy31415 
I agree with you that the part declared as Optional<bool> is strange. However, I don't know the exact reason because I didn't create the initial code.

This code is affected by all commissioners that use src/controller. However, this issue cannot occur except on platforms that use Java.
  - In the case of chip-tool, the program is terminated after commissioning, so set it again.
  - In the case of darwin, only the method set before commissioning is possible.

I don't understand exactly what "remove conditional from the setters" means. Please explain again what it means.

Reply:
@andy31415 
The implementation has been modified only for Android Platform. Please review again.